core/adapters/random.go

```diff
@@ -56,33 +69,76 @@ func (ra *Random) TaskType() models.TaskType {
 
 // Perform returns the the proof for the VRF output given seed, or an error.
 func (ra *Random) Perform(input models.RunInput, store *store.Store) models.RunOutput {
-	key, err := getKey(ra, input)
+	key, preSeed, blockHash, blockNum, err := getInputs(ra, input, store)
 	if err != nil {
-		return models.NewRunOutputError(errors.Wrapf(err, "bad key for vrf task"))
-	}
-	seed, err := getSeed(input)
-	if err != nil {
-		return models.NewRunOutputError(errors.Wrap(err, "bad seed for vrf task"))
+		return models.NewRunOutputError(err)
 	}
-	solidityProof, err := store.VRFKeyStore.GenerateProof(key, seed)
+	solidityProof, err := store.VRFKeyStore.GenerateProof(key, preSeed,
```

Comment:
dereference without nil check. Why use pointers at all?

Reply:
Thanks, I removed most of the pointers from the VRF code. Secret keys are still pointers, because you don't want to be copying those around willy-nilly.